test/image_test.py

```diff
@@ -271,11 +264,9 @@ def test_image_uv_python_requirements_error():
 
 
 def test_run_commands_secrets_type_validation(builder_version, servicer, client):
-    app = App()
-    image = Image.debian_slim().run_commands(
-        "echo hi", secrets=[Secret.from_dict({"xyz": "123"}), Secret.from_name("foo")]
-    )
-    app.function(image=image)(dummy)
+    # Happy case
+    Image.debian_slim().run_commands("echo hi", secrets=[Secret.from_dict({"xyz": "123"}), Secret.from_name("foo")])
```

Comment:
Should there be a `image.build` here to actually build the image?

Reply:
Not really needed since we do this validation eagerly (see the error case below). Maybe good to throw one on though.